packages/stream_video/lib/src/stream_video.dart

```diff
@@ -173,46 +189,83 @@ class StreamVideo {
   /// Invoked when a call was created by another user with ringing set as True.
   void Function(Call)? onIncomingCall;
 
-  /// Connects the [user] to the Stream Video service using a [TokenProvider].
-  /// This method is great if your service requires tokens to be refreshed.
-  ///
-  /// For applications outside of test and development, this method may be more
-  /// suitable as token refreshes are handled for you.
-  Future<Result<String>> connectUserWithProvider(
-    UserInfo user, {
-    required TokenProvider tokenProvider,
-  }) async {
-    _logger.i(() => '[connectUser] user.id : ${user.id}');
-    if (currentUser != null) {
-      _logger.w(() => '[connectUser] rejected (already set): $currentUser');
-      final res = await _tokenManager.getToken();
-      return res.map((data) => data.rawValue);
+  /// Connects the user to the Stream Video service.
+  Future<Result<None>> connect() async {
+    _logger.i(() => '[connect] no args');
+    // final guest user will be updated when token gets fetched
+    final tokenResult = await _waitForToken();
+    if (tokenResult is Failure) {
+      _logger.e(() => '[connect] token fetching failed: $tokenResult');
+      return tokenResult;
     }
-    final tokenResult = await _tokenManager.setTokenProvider(
-      user.id,
-      tokenProvider: tokenProvider,
-    );
-    if (tokenResult.isFailure) {
-      return tokenResult.map((data) => data.rawValue);
+    final user = _state.currentUser.valueOrNull;
+    if (user == null) {
+      _logger.w(() => '[connect] rejected (currentUser is null)');
+      return Result.error('User is not set');
     }
-    _state.currentUser.value = user;
+    _logger.v(() => '[connect] currentUser.id : ${user.id}');
 
     try {
-      final result = await _client.connectUser(user);
-      _logger.v(() => '[connectUser] completed: $result');
+      final result = await _client.connectUser(user.info);
+      _logger.v(() => '[connect] completed: $result');
       if (result is Failure) {
         return result;
       }
       _subscriptions.add(_idEvents, _client.events.listen(_onEvent));
       _subscriptions.add(_idAppState, lifecycle.appState.listen(_onAppState));
-      await pushNotificationManager?.onUserLoggedIn();
-      return tokenResult.map((data) => data.rawValue);
+      try {
+        await pushNotificationManager?.onUserLoggedIn();
+      } catch (e, stk) {
+        _logger.w(() => '[connect] #pnManager; failed: $e, $stk');
+      }
+      return const Result.success(none);
+    } catch (e, stk) {
+      _logger.e(() => '[connect] failed(${user.id}): $e');
+      return Result.failure(VideoErrors.compose(e, stk));
+    }
+  }
```

Comment:
I think the `connect` call should store the future and return with the previous one if already called. Otherwise calling connect will add subscriptions again and will also register device on every call.